modules/api/pkg/ee/resource-quota/handler.go

```diff
@@ -172,35 +214,373 @@ func GetResourceQuota(ctx context.Context, request interface{}, provider provide
 func GetResourceQuotaForProject(ctx context.Context, request interface{}, projectProvider provider.ProjectProvider,
 	privilegedProjectProvider provider.PrivilegedProjectProvider, userInfoGetter provider.UserInfoGetter,
 	quotaProvider provider.ResourceQuotaProvider) (*apiv2.ResourceQuota, error) {
-	req, ok := request.(common.GetProjectRq)
+	projectResourceQuota, projectName, err := getProjectResourceQuota(ctx, request, projectProvider, privilegedProjectProvider, userInfoGetter, quotaProvider)
+	if err != nil {
+		return nil, err
+	}
+
+	if projectResourceQuota == nil {
+		// ResourceQuota not found. Return an empty response.
+		return nil, nil
+	}
+
+	return convertToAPIStruct(projectResourceQuota, projectName), nil
+}
+
+func CalculateResourceQuotaUpdateForProject(ctx context.Context, request interface{}, projectProvider provider.ProjectProvider,
+	privilegedProjectProvider provider.PrivilegedProjectProvider, userInfoGetter provider.UserInfoGetter,
+	quotaProvider provider.ResourceQuotaProvider) (*apiv2.ResourceQuotaUpdateCalculation, error) {
+	req, ok := request.(calculateProjectResourceQuotaUpdate)
 	if !ok {
 		return nil, utilerrors.NewBadRequest("invalid request")
 	}
+
+	projectResourceQuota, projectName, err := getProjectResourceQuota(ctx, req.GetProjectRq, projectProvider, privilegedProjectProvider, userInfoGetter, quotaProvider)
+	if err != nil {
+		return nil, err
+	}
+
+	if projectResourceQuota == nil {
+		// ResourceQuota not found. Return an empty response.
+		return nil, nil
```

Comment:
From what I see, this code probably never executes. In case projectResourceQuota is not found kube returns 404 error (in the `getProjectResourceQuota` function)